flocker/provision/_install.py

```diff
@@ -825,12 +847,25 @@ def task_enable_flocker_control(distribution):
             ),
             run("echo 'flocker-control-api\t4523/tcp\t\t\t# Flocker Control API port' >> /etc/services"),  # noqa
             run("echo 'flocker-control-agent\t4524/tcp\t\t\t# Flocker Control Agent port' >> /etc/services"),  # noqa
-            run_from_args(['service', 'flocker-control', 'start']),
+            run_from_args(['service', 'flocker-control', action.lower()]),
         ])
+
     else:
         raise DistributionNotSupported(distribution=distribution)
 
 
+def validate_start_action(action):
+    """
+    validates if the action given is a valid one  - currently only
+    start and restart are supported
+    """
+    valid_actions = ["start", "restart"]
+    for valid_action in valid_actions:
+        if action.lower() == valid_action:
+            return
```

Comment:
You can condense this just to:

```python
if action.lower() in valid_actions:
    return
```

Once that's done, then it probably makes more sense to invert the conditional and drop the explicit return, e.g.:

```python
if action.lower() not in valid_actions:
    raise UnknownAction(action)
```